src/main/java/model/DeadLineTask.java

```diff
@@ -0,0 +1,34 @@
+package main.java.model;
+
+import main.java.model.Task;
+
+import main.java.exceptions.NoDescriptionException;
+
+public class DeadLineTask extends Task {
+    protected String by;
+
+    public DeadLineTask() {
+        super("deadline task");
+    }
+
+    public DeadLineTask(String description, String by) throws NoDescriptionException {
+        super(description, "deadline task");
+        this.by = by;
+    }
+
+    public DeadLineTask(String... params) throws NoDescriptionException {
+        super(params[0], "deadline task");
+        this.by = params[1];
+    }
```

Comment:
Why did you chose the var args approach if your code body is going to use only 2 arguments. Would rather create the method like 

`public DeadLineTask(String A, String B) ...` with proper naming for A / B.

This adds clarity when someone else reads the code. He can understand what to expect for the input parameters, rather than trying to infer what params[0] / params[1] is

Reply:
Agree with Johan, using var args when you have a fixed number of parameter seems unnecessary and could confuse the reader.